db/migrations/1657379568_create_courses_table.up.sql

```diff
@@ -16,9 +16,11 @@ CREATE TABLE courses
     deleted_at      timestamp
 );
 
-CREATE UNIQUE INDEX courses_id_uindex
-    ON courses (id);
+COMMENT ON COLUMN courses.deleted_at IS 'Timestamp indicating when a course was softly deleted, allowing for data recovery and historical queries without permanently removing the record from the database. A NULL value means the subscription is active.';
+
 CREATE UNIQUE INDEX courses_uuid_uindex
     ON courses (uuid);
+CREATE UNIQUE INDEX courses_code_uindex
+    ON courses (code, deleted_at) NULLS NOT DISTINCT;
```

Comment:
With this we will never be able to unarchive a course, do we really want to do that?